### ISSUE TITLE
Send webhook based on field value change of model. 

### ISSUE THREAD

**Issue by wtzr** [MrThearMan/django-signal-webhooks#1]
Hi,

Works like a charm, but wondering how i could best implement the following:

Is there a way to save a specific webhook per field of a model. For example:

I have a transaction model, that has an options field that can be "expired", "pending", "completed". 

I would like to setup specific webhooks when:

- status has changed to exipred
- status has changed to pending
- status has changed to completed. 

Thanks

**Comment by wtzr** [comment]
ah got it. That worked well. 

One more question :)

Is there also a way to NOT send a webhook, when i'm using webhook_data()  ?

I only want to send a webhook on a model UPDATE if the transaction belongs to an organization. I currently achieve this be checking if the webhook.url corresponds with endpoint.url. However, if that statement is false, it will still send an empty webhook. How do i make sure that webhook is not send?

Thanks!

**Comment by wtzr** [comment]
I can confirm this works perfectly. Thanks for that release!

**Comment by wtzr** [comment]
One more thing ;)

We want to use this for our customers, so they can get webhooks from our application. However... they should only receive the webhooks from their organization. 

I tried it with setting up different webhooks on the same model in the backend like this:

Customer1:
models: api.models
signal: update
endpoint url: www.xxx.com

Customer2:
models: api.models
signal: update
endpoint url: www.yyy.com

And the custom code:

```
        webhook = Webhook.objects.get(endpoint=self.organization.payload_url)

        if webhook.endpoint == self.organization.payload_url:
            
            return {
                'order_id': self.order_id,
                'order_number': self.order_number,
                'first_name': self.first_name,
            }
        else:
            raise WebhookCancelled("Transaction not completed")
```

However, that won't work. Because one of the webhooks will always fire and send the webhook to both organizations. Perhaps i'm missing something obvious, but i can't figure out to split this per customer.

**Comment by MrThearMan** [comment]
Hmm, you might need to write a custom `post_save_handler` ([see the default one here](https://github.com/MrThearMan/django-signal-webhooks/blob/main/signal_webhooks/handlers.py#L125)) to filter the webhooks that get fired based on the data you get from `.webhook_data()`. Then you point to your handler in the project settings:

```python
SIGNAL_WEBHOOKS = {
    "HOOKS": {
        "path.to.my.model": {
            "CREATE": ...,   # default handling
            "UPDATE": "path.to.my.handler",  # custom handling
            "DELETE": ...,   # default handling
        },
    },
}
```

It's not ideal, as you might need to duplicate a lot of the code in the default handler. I'll make a note on this, maybe I can think of a good abstraction level where the filtering rules could be placed. Tell me how it works out!

ps. I'll close the issue as the original problem as been solved, but feel free to keep commenting if you have any questions

**Comment by MrThearMan** [comment]
Hi, I simplified the inner workings of the library a bit and added an abstraction level for filtering the webhooks that are fired per-model, per-method. You can now define a [FILTER_KWARGS](https://github.com/MrThearMan/django-signal-webhooks/blob/98bc5ad1c6e012ac3fee3c5d3e4c1b232a6b9fa2/signal_webhooks/settings.py#L57) setting for a custom function that adds keyword arguments [to the QuerySet.filer() used to fetch the hooks](https://github.com/MrThearMan/django-signal-webhooks/blob/98bc5ad1c6e012ac3fee3c5d3e4c1b232a6b9fa2/signal_webhooks/models.py#L21). In you case, you could do something like this:

```python
# /myproject/settings.py

SIGNAL_WEBHOOKS = {
    ...
    "FILTER_KWARGS": "path.to.file.my_filter_handler",
    ...
}
```

```python
# /path/to/file.py

def my_filter_handler(instance: Model, method: Literal["CREATE", "UPDATE", "DELETE"]) -> Dict[str, Any]:
    if isinstance(instance, Customer) and method == "UPDATE":
        return {"endpoint": instance.organization.payload_url}
    return {}
```

This has been released in [v0.2.4](https://github.com/MrThearMan/django-signal-webhooks/releases/tag/v0.2.4). If you decide to update, you'll need to adapt your custom hook handler (if you decided to write one based on my previous comment) to the [new function signature](https://github.com/MrThearMan/django-signal-webhooks/blob/98bc5ad1c6e012ac3fee3c5d3e4c1b232a6b9fa2/signal_webhooks/handlers.py#L111) or use the new abstraction level with FILTER_KWARGS.
